lhotse/recipes/radio.py

```diff
@@ -0,0 +1,144 @@
+"""
+This recipe prepares data collected from radio streamed on the web. The data
+have some metadata attached to them, including the geographic location of
+broadcast, date and time of the recorded clip, as well as a unique station
+identifier.
+
+Obtaining the data
+-----------------------------------------------------------
+If you want to use this corpus please email: wiesner@jhu.edu
+
+As the data are collected from radio stream, they cannot be broadly
+disseminated or used for commercial purposes. In the email, include your
+affiliated academic institution and the intended use for the data and we will
+the data to you if it is indeed for non-commercial, academic purporses.
+
+Description
+------------------------------------------------------------
+The data consist of ∼4000 hours of speech collected between
+September 27, 2023 to October 1, 2023, in 9449 locations all over the world,
+from 17171 stations. 
+
+These data were used for Geolocation of speech in order to answer the question,
+Where are you from? in the paper 
+
+Where are you from? Geolocating Speech and Applications to Language
+Identification, presented at NAACL 2024. Please read for a full descrption
+and please cite as 
+ 
+@inproceedings{foley2024you,
+  title={Where are you from? Geolocating Speech and Applications to Language Identification},
+  author={Foley, Patrick and Wiesner, Matthew and Odoom, Bismarck and Perera, Leibny Paola Garcia and Murray, Kenton and Koehn, Philipp},
+  booktitle={Proceedings of the 2024 Conference of the North American Chapter of the Association for Computational Linguistics: Human Language Technologies (Volume 1: Long Papers)},
+  pages={5114--5126},
+  year={2024}
+}
+"""
+import json
+import logging
+from pathlib import Path
+from typing import Dict, Optional, Union, NamedTuple, Tuple, List
+from tqdm import tqdm
+from lhotse.audio import Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike
+import lhotse
+import re
+
+from functools import partial
+
+from lhotse.parallel import parallel_map
+from lhotse.audio import set_ffmpeg_torchaudio_info_enabled
+import torchaudio
```

Comment:
Make torchaudio imports local, it’s not a core dependency anymore.

Reply:
I think it should all be fixed now. I have an associated recipe in icefall for doing LID and geolocation that I am going to submit a pull request for. Part of that involves a custom dataset class, but it's fairly specific to this one task. I assume it would be better to just have that as part of the icefall recipe and not part of Lhotse?

Reply:
Yeah that makes sense.